src/Components/PaymentFilter/PayoneBNPLPaymentMethodFilter.php

```diff
@@ -4,21 +4,51 @@
 
 namespace PayonePayment\Components\PaymentFilter;
 
+use PayonePayment\Components\ConfigReader\ConfigReader;
 use PayonePayment\Core\Utils\AddressCompare;
 use Shopware\Core\Checkout\Customer\Aggregate\CustomerAddress\CustomerAddressEntity;
 use Shopware\Core\Checkout\Order\Aggregate\OrderAddress\OrderAddressEntity;
 use Shopware\Core\Checkout\Payment\PaymentMethodCollection;
+use Shopware\Core\System\SystemConfig\SystemConfigService;
 
 class PayoneBNPLPaymentMethodFilter extends DefaultPaymentFilterService
 {
+    public function __construct(
+        private readonly SystemConfigService $systemConfigService,
+        private readonly string $paymentHandlerClass,
+        ?array $allowedCountries = null,
+        ?array $allowedB2bCountries = null,
+        ?array $allowedCurrencies = null,
+        float $allowedMinValue = 0.0,
+        ?float $allowedMaxValue = null,
+    ) {
+        parent::__construct(
+            $paymentHandlerClass,
+            $allowedCountries,
+            $allowedB2bCountries,
+            $allowedCurrencies,
+            $allowedMinValue,
+            $allowedMaxValue
+        );
+    }
+
     public function filterPaymentMethods(PaymentMethodCollection $methodCollection, PaymentFilterContext $filterContext): PaymentMethodCollection
     {
         $methodCollection = parent::filterPaymentMethods($methodCollection, $filterContext);
 
+        $differentShippingAddressAllowed = (bool) $this->systemConfigService->get(
```

Comment:
i think it makes sense, that this would be configurable via DI parameters, because the addresses can be different only on invoice payment. (not installment, direct debit and so on)

Maybe it would be also a good idea to move this into the default-payment-filter and make it configurable. 
So we could use it in the future for other payment methods.